unitTests/ui/QuerySuggestAddonTest.ts

```diff
@@ -188,5 +188,45 @@ export function QuerySuggestAddonTest() {
         done();
       });
     });
+
+    describe('when the characterThresholdForSuggestions option is set to a value', () => {
+      let magicBoxText;
+      let querySuggestSuccessSpy;
+      beforeEach(() => {
+        simulateFakeSuggestions({ completions: [{ executableConfidence: 1, expression: 'foo', highlighted: 'foo', score: 1 }] });
+        magicBoxText = jasmine.createSpy('magicBoxText');
+        querySuggestSuccessSpy = jasmine.createSpy('spyOnEvent');
+        omnibox.cmp.options.characterThresholdForSuggestions = 3;
+        omnibox.cmp.magicBox.getText = magicBoxText;
+        $$(omnibox.env.root).on(OmniboxEvents.querySuggestSuccess, () => querySuggestSuccessSpy());
+      });
+
+      it('when the magic box text length is equal to the minimumNumberOfCharactersForSuggestions option should trigger query suggestion', async done => {
+        magicBoxText.and.returnValue('foo');
+        querySuggest = new QuerySuggestAddon(omnibox.cmp);
+        await querySuggest.getSuggestion();
+        expect(magicBoxText).toHaveBeenCalled();
+        expect(querySuggestSuccessSpy).toHaveBeenCalled();
```

Comment:
I think it would be better to look at the array returned by `querySuggest.getSuggestion();`.

The test would instead check that : 
```
const suggestionsReturned = await querySuggest.getSuggestion();
expect(suggestionsReturned.length).toBe(1)
```

and when you wanna check that it should not return anything/skip query suggestions : 

```
const suggestionsReturned = await querySuggest.getSuggestion();
expect(suggestionsReturned.length).toBe(0)
```

This goes for all your tests.

Reply:
Great to know.
I'll do that